types/time.go

```diff
@@ -1737,10 +1732,116 @@ func ExtractTimeValue(unit string, format string) (int64, int64, int64, float64,
 	case "YEAR_MONTH":
 		return extractTimeValue(format, MonthIndex, YearMonthMaxCnt)
 	default:
-		return 0, 0, 0, 0, errors.Errorf("invalid singel timeunit - %s", unit)
+		return 0, 0, 0, 0, errors.Errorf("invalid single timeunit - %s", unit)
 	}
 }
 
+// ExtractDurationValue extract the value from format to Duration.
+func ExtractDurationValue(unit string, format string) (Duration, error) {
+	unit = strings.ToUpper(unit)
+	switch unit {
+	case "MICROSECOND", "SECOND", "MINUTE", "HOUR", "DAY", "WEEK", "MONTH", "QUARTER", "YEAR":
+		return extractSingleDurationValue(unit, format)
+	case "SECOND_MICROSECOND":
+		_, _, _, d, err := extractTimeValue(format, MicrosecondIndex, SecondMicrosecondMaxCnt)
```

Comment:
After this PR(#10329), large `duration` will be rounded up to `day`, so it's better to keep and check all unit fields.

Reply:
i'll change codes when that one merged.